driver-async/src/main/com/mongodb/async/client/internal/CommandMarker.java

```diff
@@ -79,13 +79,18 @@ public void onResult(final RawBsonDocument result, final Throwable t) {
                 }
             }
         };
-        executeCommand(databaseName, command, new SingleResultCallback<RawBsonDocument>() {
+        runCommand(databaseName, command, new SingleResultCallback<RawBsonDocument>() {
             @Override
             public void onResult(final RawBsonDocument result, final Throwable t) {
                 if (t == null) {
                     wrappedCallback.onResult(result, null);
                 } else if (t instanceof MongoTimeoutException && processBuilder != null) {
-                    executeCommand(databaseName, command, wrappedCallback);
+                    try {
```

Comment:
I would rather the exception is passed via the callback, instead of being thrown as that follows the normal async execution control flow.

Reply:
I'm not sure I understand.  Exceptions from startProcess will be caught on line 91 and passed via the callback.

Reply:
previously errors were passed back via callbacks, instead of being caught and thrown. Using callbacks is more idiomatic for async.

Reply:
I pushed a new patch that used callbacks.  I find the code harder to understand this way, but perhaps I misunderstood your intention.  Let me know what you think.

Reply:
This follows async idioms better - in the past bugs have occurred because mixing the two idioms.  This is very similar to the original code now - apart from the initial starting of the process builder.

Only thing I can see is you no longer need the try block surrounding `startProcessAndContinue`.

Reply:
I'd like to talk about this idiom further, as i don't understand it.  Generally I thought we should avoid using callbacks when we know we're doing something synchronous, and use callbacks where we at least might be doing something asynchronous.  

But until then, I'll go with the suggestion.

With regard to the try block, I'm unsure.  I see in many other places in the driver there are try blocks around the entire callback to guard against unexpected exceptions thrown before doing something asynchronous, and having those exceptions unexpectedly lost in the ether.

Reply:
I think its safe to remove the try catch as `startProcessAndContinue` already contains a try and catch block and runCommand should already be protected and internally it uses a `ErrorHandlingResultCallback`

Reply:
Done, and also changed the catch block surrounding ProcessBuilder.start, because that method can throw RuntimeException (like a SecurityException) in addition to IOException